test/unit/object_geometry.js

```diff
@@ -802,4 +802,38 @@
     assert.deepEqual(coords[3].y, 85, 'return bottom left with skewY skewX angle Y');
   });
 
+  QUnit.test('isPartiallyOnScreen', function(assert) {
+    var cObj = new fabric.Object(
+      { left: -10, top: -10, width: canvas.getWidth() + 100, height: canvas.getHeight(), strokeWidth: 0});
+    canvas.viewportTransform = [1, 0, 0, 1, 0, 0];
+    cObj.canvas = canvas;
+    cObj.setCoords();
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), true, 'object is partially on screen because it include the canvas');
+    cObj.left = 100;
+    cObj.top = 100;
+    cObj.setCoords();
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), false, 'object is not partially on screen because it include the canvas');
+    cObj.left = -1000;
+    cObj.top = -1000;
+    cObj.setCoords();
+    assert.equal(cObj.isPartiallyOnScreen(cObj.aCoords), false, 'object is not partially on screen because it is outside the canvas');
+  });
```

Comment:
we have more test to cover and edge cases i believe. Intersect can return true for coincidents segments too.